packages/cardpay-sdk/sdk/safes/base.ts

```diff
@@ -267,24 +281,40 @@ export default class Safes {
       estimate.gasToken,
       ZERO_ADDRESS
     );
+
+    let txnHash = result.ethereumTx.txHash;
+
     if (typeof onTxnHash === 'function') {
-      await onTxnHash(result.ethereumTx.txHash);
+      await onTxnHash(txnHash);
     }
-    return await waitUntilTransactionMined(this.layer2Web3, result.ethereumTx.txHash);
+    return await waitUntilTransactionMined(this.layer2Web3, txnHash);
   }
 
+  async setSupplierInfoDID(txnHash: string): Promise<TransactionReceipt>;
   async setSupplierInfoDID(
     safeAddress: string,
     infoDID: string,
     gasToken: string,
     txnOptions?: TransactionOptions,
     contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt>;
+  async setSupplierInfoDID(
+    safeAddressOrTxnHash: string,
+    infoDID?: string,
+    gasToken?: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
   ): Promise<TransactionReceipt> {
+    if (!infoDID) {
```

Comment:
ditto with the comments above about testing whether the value you actually care about is shaped like an address or txnHash instead of testing a side effect